pkg/snapstore/swift_snapstore_test.go

```diff
@@ -150,3 +178,50 @@ func handleDeleteObject(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusNotFound)
 	}
 }
+
+// handleBulkDeleteObject creates an HTTP handler at `/testContainer/testObject` on the test handler mux that
+// responds with a `Delete` response.
+func handleBulkDeleteObject(w http.ResponseWriter, r *http.Request) {
+	objectMapMutex.Lock()
+	defer objectMapMutex.Unlock()
+
+	var jsonResponse bulkDeleteJSONResponse
+
+	buf := new(bytes.Buffer)
+	if _, err := io.Copy(buf, r.Body); err != nil {
+		errorMessage := fmt.Sprintf("failed to read content %v", err)
+		logrus.Errorf(errorMessage)
+		jsonResponse.ErrorStrings = append(jsonResponse.ErrorStrings, errorMessage)
+		w.WriteHeader(http.StatusOK)
+		return
+	}
+
+	segmentObjects := strings.Split(string(buf.Bytes()), "\n")
+	// last string is an empty string, artefact of '\n' at the end
+	segmentObjects = segmentObjects[:len(segmentObjects)-1]
```

Comment:
Thanks for the suggestion, makes the segment object list parsing from the buffer cleaner!

From the docs of [DLOs](https://docs.openstack.org/swift/latest/overview_large_objects.html#direct-api):

> The manifest file is simply a zero-byte (not enforced) file with the extra X-Object-Manifest: <container>/<prefix> header, where <container> is the container the object segments are in and <prefix> is the common prefix for all the segments.

> The manifest file should have no content. However, this is not enforced. If the manifest path itself conforms to container/prefix specified in X-Object-Manifest, and if manifest has some content/data in it, it would also be considered as segment and manifest’s content will be part of the concatenated GET response. The order of concatenation follows the usual DLO logic which is - the order of concatenation adheres to order returned when segment names are sorted.

The segment object names are not added to the manifest object contents. The segment object files that are to be deleted are inferred from the name of the manifest object file.
Any contents that are written to the manifest object will make it be treated as a segment object too, which is not intended behavior.